### DISCUSSION TITLE
_ITestOutputHelper in xunit.v3.core?

### DISCUSSION THREAD

**Discussion by virzak** [xunit/xunit#2423] [Question] [answered]
In 38242c03 the `_ITestOutputHelper`  interface moved from xunit.v3.common to xunit.v3.core

A project that uses this class and previously [targeted](https://github.com/trbenning/serilog-sinks-xunit/blob/c2e334fa840ccd290aba0ed8e0831b54ccebd99d/src/Serilog.Sinks.XUnit/Serilog.Sinks.XUnit.csproj#L12) netstandard can no longer be upgraded to netstandard2.0. Instead it must target .NETCoreApp 3.1 or .NETFramework 4.7.2.

Wonder if that's what has to happen or if there's a way of targeting netstandard2.0.

**Comment by bradwilson** [comment, marked as answer]
The short answer is: there is no way to have a unit test project target `netstandard2.0`, so if you're asking about unit tests, then you must choose one of those two things (.NET Framework >= 4.72 or .NET Core >= 3.1, which of course includes .NET >= 5.0). This doc page (["Why doesn't xUnit.net support netstandard?"](https://xunit.net/docs/why-no-netstandard)) is slightly out of date w/rt v3 but still applicable. In fact, it should be even more obvious now with the v3 architecture that projects MUST be executable; there is no such thing as an executable netstandard project.

If you mean you want to use this from an extensibility project targeting `xunit.v3.core`, the current NuGet targets include `net472` and `netcoreapp3.1`, even though `xunit.v3.core.dll` itself is built against `netstandard2.0`, precisely to drive home the point I make above w/rt projects not supporting netstandard. If you're doing extensibility, you should instead depend on `xunit.v3.extensibility.core`, which allows you to target `netstandard2.0` (so you're not forced to multi-target). The `xunit.v3.core` NuGet package depends on `xunit.v3.extensibility.core` NuGet package; the DLLs themselves live in `xunit.v3.extensibility.core`.

Make sense?

**Comment by virzak** [comment]
`xunit.v3.extensibility.core` was the key. As soon as it was referenced, everything compiled. It was actually in the v3 docs too. Thanks for the answer.
